src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -355,7 +355,10 @@ private void setMinimumLogLevel(String minimumLogLevel, ComponentLogConfiguratio
 
     private void setDiskSpaceLimit(String diskSpaceLimit, String diskSpaceLimitUnit,
                                    ComponentLogConfiguration componentLogConfiguration) {
-        if (!StringUtils.isEmpty(diskSpaceLimit) && !StringUtils.isEmpty(diskSpaceLimitUnit)) {
+        if (!StringUtils.isEmpty(diskSpaceLimit)) {
```

Comment:
Is `diskSpaceLimitUnit` still nonempty when there is no node change for `DISK_SPACE_LIMIT_UNIT_CONFIG_TOPIC_NAME`? Suppose the initial disk space limit is 20MB, if customer updates it to 30MB, so that the `diskSpaceLimitUnit` is not updated, will it be changed to 30KB since the default is KB?

Reply:
The 2 configs are each set independently from their value in the input map. So if the user changes the diskSpaceLimit but does not change the diskSpaceLimitUnit, then the units will remain the same. In the example you proposed, the disk space limit will change from 20MB to 30MB.  The units will only change to KB if the diskSpaceLimitUnit value is explicitly removed